test/k8sT/Services.go

```diff
@@ -439,6 +439,127 @@ var _ = Describe("K8sServicesTest", func() {
 			doRequestsFromThirdHostWithLocalPort(url, count, checkSourceIP, 0)
 		}
 
+		getPodOnNode := func(nodeName string, podFilter string) (string, string) {
+			var podName string
+			podsNodes, err := kubectl.GetPodsNodes(helpers.DefaultNamespace, fmt.Sprintf("-l %s", podFilter))
+			ExpectWithOffset(2, err).Should(BeNil(), "Cannot retrieve pods nodes with filter %q", podFilter)
+			Expect(podsNodes).ShouldNot(BeEmpty(), "No pod found in namespace %s with filter %q", helpers.DefaultNamespace, podFilter)
+			for pod, node := range podsNodes {
+				if node == nodeName {
+					podName = pod
+					break
+				}
+			}
+			ExpectWithOffset(2, podName).ShouldNot(BeEmpty(), "Cannot retrieve pod on node %s with filter %q", nodeName, podFilter)
+			podsIPs, err := kubectl.GetPodsIPs(helpers.DefaultNamespace, podFilter)
+			ExpectWithOffset(2, err).Should(BeNil(), "Cannot retrieve pods IPs with filter %q", podFilter)
+			Expect(podsIPs).ShouldNot(BeEmpty(), "No pod IP found in namespace %s with filter %q", helpers.DefaultNamespace, podFilter)
+			podIP := podsIPs[podName]
+			return podName, podIP
+		}
+
+		// srcPod:      Name of pod sending the datagram
+		// srcIP:       IPv4 of pod sending the datagram
+		// fromNode:    Node hosting the pod sending the datagram
+		// dstPodIP:    Receiver pod IP (for checking in CT table)
+		// dstPodPort:  Receiver pod port (for checking in CT table)
+		// dstIP:       Target endpoint IP for sending the datagram
+		// dstPort:     Target endpoint port for sending the datagram
+		doFragmentedRequest := func(srcPod string, srcIP string, fromNode string, dstPodPort int, dstIP string, dstPort int32) {
+			var (
+				blockSize  = 512
+				blockCount = 10
+				srcPort    = 12345
+			)
+			ciliumPodK8s1, err := kubectl.GetCiliumPodOnNode(helpers.CiliumNamespace, helpers.K8s1)
+			Expect(err).Should(BeNil(), fmt.Sprintf("Cannot get cilium pod on k8s1"))
+			ciliumPodK8s2, err := kubectl.GetCiliumPodOnNode(helpers.CiliumNamespace, helpers.K8s2)
+			Expect(err).Should(BeNil(), fmt.Sprintf("Cannot get cilium pod on k8s2"))
+
+			ciliumPod := ciliumPodK8s1
+			if fromNode == helpers.K8s2 {
+				ciliumPod = ciliumPodK8s2
+			}
+
+			_, dstPodIPK8s1 := getPodOnNode(helpers.K8s1, testDS)
+			_, dstPodIPK8s2 := getPodOnNode(helpers.K8s2, testDS)
+
+			// Get initial number of packets for the flow we test
+			// from conntrack table. The flow is probably not in
+			// the table the first time we check, so do not stop if
+			// Atoi() throws an error and simply consider we have 0
+			// packets.
+
+			// Field #7 is "RxPackets=<n>"
+			cmdIn := "cilium bpf ct list global | awk '/%s/ { sub(\".*=\",\"\", $7); print $7 }'"
```

Comment:
We have JSON encoding to make this kind of call more resilient.

The CT list can use:
```
cilium bpf ct list global -o json
```

I believe that if this is just filtering the RxPackets field and *only* printing that, the equivalent jq filter is:

```
jq '.MapValue.RxPackets'
```

This is not blocking this PR but it would be nice to follow up after this PR to fix this kind of case up just in case we change the standard CLI format in future. Same for the other calls in this test.

Reply:
Right, I didn't realise we had JSON here.

... On the other hand, JSON output for ct map is currently broken (#2976). Oh well. I'll fix in a separate PR, and update here as a follow-up. I saw cilium also has a `-o jsonpath` option, but it seems less flexible than `jq`, I'll see what works best when doing the follow-up.

Reply:
Thanks for digging into this, I didn't realise it was still spitting out the wrong output.

re: jsonpath vs. jq, I didn't think `-o jsonpath` was supported for `cilium bpf ct list`. We use jsonpath a lot in the tests and unfortunately the syntax isn't interchangeable but *shrug* whatever works for you IMO.